### PR TITLE
 Clarify error for AssemblyVersion with wildcard when deterministic VB

### LINKED ISSUE #30738: Confusing error message about determinism and AssemblyVersion in VB
Recently updated to Visual Studio 2017 v 15.8.5

Changing AssemblyInfo.vb
from default (swapping comments):

```
' You can specify all the values or you can default the Build and Revision Numbers
' by using the '*' as shown below:
' <Assembly: AssemblyVersion("1.0.*")>
<Assembly: AssemblyVersion("1.0.0.0")>
<Assembly: AssemblyFileVersion("1.0.0.0")>
```

to:

```
<Assembly: AssemblyVersion("1.0.*")>
'<Assembly: AssemblyVersion("1.0.0.0")>
'<Assembly: AssemblyFileVersion("1.0.0.0")>
```

gives the following error:

Error BC36962 The specified version string does not conform to the required format - major[.minor[.build|*[.revision|*]]]

_This issue has been moved from https://developercommunity.visualstudio.com/content/problem/346841/bc36962.html
VSTS ticketId: 694617_
_These are the original issue comments:_
(no comments)
_These are the original issue solutions:_
(no solutions)

### PR DESCRIPTION
**Pull request by A9G-Data-Droid**
Translate CSharp solution to VB from PR #22973

 - Fix path to VisualBasicLspBuildOnlyDiagnostics in comment
 - Fixes #30738

KNOWN ISSUE:
This has generated a series of `needs-review-translation` flags in other language files. Should I make an attempt at updating languages I do not speak? Or leave this to go through a translation process?

### CODE REVIEW

**Comment by A9G-Data-Droid** [comment]
@dotnet-policy-service agree

File: src/Compilers/VisualBasic/Portable/Symbols/Source/SourceAssemblySymbol.vb [hunk 1]
```diff
@@ -1042,8 +1042,14 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Symbols
                 Dim verString = DirectCast(attrData.CommonConstructorArguments(0).ValueInternal, String)
                 Dim version As Version = Nothing
                 If Not VersionHelper.TryParseAssemblyVersion(verString, allowWildcard:=Not _compilation.IsEmitDeterministic, version:=version) Then
-                    diagnostics.Add(ERRID.ERR_InvalidVersionFormat, GetAssemblyAttributeFirstArgumentLocation(arguments.AttributeSyntaxOpt))
+                    Dim attributeArgumentSyntaxLocation As Location = GetAssemblyAttributeFirstArgumentLocation(arguments.AttributeSyntaxOpt)
+                    If _compilation.IsEmitDeterministic AndAlso verString.Contains("*"c) Then
```
Comment (jaredpar) [inline src/Compilers/VisualBasic/Portable/Symbols/Source/SourceAssemblySymbol.vb:1046]: The C# equivalent considers that `verString` can be `null`. Should replicate that check here or attempt to remove it in C#

**Comment by 333fred** [comment]
> Should I make an attempt at updating languages I do not speak? Or leave this to go through a translation process?

Leave them as is. Our translation team updates these regularly.

File: src/Compilers/VisualBasic/Portable/Symbols/Source/SourceAssemblySymbol.vb (hunk 1 above)
Comment (A9G-Data-Droid) [reply src/Compilers/VisualBasic/Portable/Symbols/Source/SourceAssemblySymbol.vb]: Thanks. I have added an identical null check to remain consistent.

File: src/Compilers/VisualBasic/Portable/Errors/Errors.vb [hunk 2]
```diff
@@ -72,6 +72,8 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
         ERR_MutuallyExclusiveOptions = 2046
         ERR_BadSwitchValue = 2047
 
+        ERR_InvalidVersionFormatDeterministic = 8357
```
Comment (jaredpar) [inline src/Compilers/VisualBasic/Portable/Errors/Errors.vb:75]: @333fred do we normally try and match error codes like this? Semes like a fairly high jump

Comment (333fred) [reply src/Compilers/VisualBasic/Portable/Errors/Errors.vb:75]: No, we make no attempts to match error codes between C# and VB.

Comment (A9G-Data-Droid) [reply src/Compilers/VisualBasic/Portable/Errors/Errors.vb:75]: Thanks! That is new information to me and unexpected. Would I assign 2048? or jump to the bottom and use 37331 ?

Comment (333fred) [reply src/Compilers/VisualBasic/Portable/Errors/Errors.vb:75]: 37330 please, and increment ERR_NextAvailable to 37331.

**Review by 333fred — APPROVED** [review]

**Review by jaredpar — APPROVED** [review]

**Comment by 333fred** [comment]
Thanks @A9G-Data-Droid !

**Pull request merged**